### PR TITLE
This extends PR from issue: #21

### PR DESCRIPTION
**Pull request by paajake**
This PR provides an extension to the PR for issue #21 and contains some improvements for the initial project setup and project dependencies management by;

- Cleaning up the requirements.txt and adding required dependencies
- Integrating a `.env` file and usage for ALL project variables configuration
- Adding a check for successful API requests
- Updates README.md

### How to Test
- Follow the updated README.md to setup the project.
- Run a project script and ensure NO errors are reported. 

### Thanks
- Special thanks to the PyCon2023 organisers, especially the sprint organisers for the opportunity and encouragement to contribute.
- Special thanks to @lwasser for the encouragement, guidance and leadership in making all this possible.
- Thanks to @bbulpett for the initial PR for issue #21.

### CODE REVIEW

**Comment by paajake** [comment]
pre-commit.ci autofix

**Comment by lwasser** [comment]
hey there @paajake THANK YOU for this pr!! wow! will look at it now. And so good to meet you at pycon us this year!

**Comment by lwasser** [comment]
@all-contributors please add @paajake for code, design, review

File: README.md [hunk 1]
```diff
@@ -3,18 +3,22 @@
 
 This repo contains a small module and some scripts that parse through GitHub repos in the pyOpenSci organization.
 
-To begin create a local environment and install this package in editable mode.
-
-`pip install -e .`
+To begin;
+1. create a local environment and activate it.
```
Comment (lwasser) [inline README.md:7]:
```suggestion
1. Create a local environment and activate it.
```

File: README.md [hunk 2]
```diff
@@ -3,18 +3,22 @@
...
+2. Install the required dependencies via the requirement.txt file by running the following command;
+`pip install -r requirements.txt`
 
 For an action to work will need to figure out the token part: https://github.com/orgs/community/discussions/46376
```
Comment (lwasser) [inline README.md:11]:
```suggestion
`pip install -r requirements.txt`

3. Install the pyos meta package in development/editable mode:

`pip install -e . `

### TODO: 
For an action to work will need to figure out the token part: https://github.com/orgs/community/discussions/46376
```

**Comment by lwasser** [comment]
pre-commit.ci autofix

**Comment by lwasser** [comment]
This is a great addition / improvement. so many thanks for it @paajake !! merging now and i've added you to our contributors list for this repo!

**Pull request merged**